BoardWebApp-Tests/HomePageTests.cs

```diff
@@ -1,44 +1,88 @@
         private readonly ITestOutputHelper _output;
-        //private string _mockAuthenticationHash;
 
         public HomePageTests(ITestOutputHelper testOutputHelper)
         {
             _output = testOutputHelper;
-            _dbContext = null;
+            _dbContext = InMemoryDb.InitInMemoryDbContext();
         }
         public void Dispose()
         {
-            _dbContext = null;
+            _dbContext.Database.EnsureDeleted();;
         }
 
+        // --------------------- This test verifes the authentication method is returning a user whenever the cookie value is matching a User record ---------------------
+        // --------------------- Maybe not the place for this test? ---------------------
         [Theory]
         [InlineData("e8107daab70f9bec25d249541eb247f36514248a14597b3cdc5ebaa3bb140a68c472a2c6627ceb7c0f1b1f1c5e8ed5a903fcede83b4b9da720697c3dc127ffff", true)] // Zack (notaracist@murica.com)
         [InlineData("08176ba0671827d033a25cfa6608d92caca8008527af3ff32dc23012dc99d554217161eda88a0d393461c215f94cdbfb787e82ed7c8e5db2a2bfcbea69c7a0d4", true)] // Trudy (trudy@board.com)
         [InlineData("247f36514248a14597b3cdc5ebaa3bb140a68c472a2c6627ceb7c0f1b1f1c5ehk7d107ebd30c26531e71747343btd42d5f64f6b423a4f9775490s281fceaadfb", false)] // not in db
         public void AuthenticationTest(string mockAuthHash, bool expectedResult)
         {
-            _dbContext = RegistrationValidationTests.InitInMemoryDbContext();
             User authenticatedUser = User.AuthenticateBasedOnCookieValue(mockAuthHash, _dbContext);
             bool actualResult = (authenticatedUser != null); // true if a user is found
 
             Assert.Equal(expectedResult, actualResult);
 
-            _dbContext = null;
+        }
+
+        // --------------------- The FOUR tests below are testing the methods retrieving the Boards and Projects a user owns and they're a member of ---------------------
+        [Theory]
+        [InlineData("notaracist@murica.com", (object)(new string[]{"Zack's shared board"}))]
+        [InlineData("trudy@board.com", (object)(new string[]{ "Trudy's board"}))]
+        [InlineData("alicia@board.com", (object)(new string[]{ /* doesn't own any boards */ }))]
+        public void GetBoardsWhereOwner(string userEmail, string[] expectedBoardNames)
+        {
+            List<User> allUsers = new List<User>(_dbContext.User);
+            User userForEmail = null;
+            foreach(User u in allUsers)
+            {
+                if(u.Email == userEmail)
+                    userForEmail = u;
+            }
+            List<string> actualBoardNames = new List<string>();
+            List<Board> actualBoards = userForEmail.GetBoardsWhereThisUserIsOwner(_dbContext);
+            foreach(Board actualBoard in actualBoards)
+            {
+                _output.WriteLine("Actual board name : " + actualBoard.BoardName);
+                actualBoardNames.Add(actualBoard.BoardName);
+            }
+
```

Comment:
Line 84 invokes the method that's being tested. In this case it returns all boards where the specified user is of type "Member".
Then the loop on line 86 is extracting the names of the retrieved boards. I found it easier as they're unique within the InMemoryDb and I found it easier to read. Could've used the Id's to verify the correct boards are returned, but that would look like a mess.
Then the assert on 92 is making sure that the returned list is the same size as the expected.
If it's not then the method failed and there's no point continuing with the rest.
Else it will verify that all the expected board names are in the returned list.